### PR TITLE
feat: add hamming distance

### PR DESCRIPTION
**Pull request by aybarsnazlica**
#### Description of Change

Add an algorithm to calculate the Hamming distance between two strings.

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: misc/hamming_distance.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
+/**
+ * @file
+ * @brief Hamming distance. https://en.wikipedia.org/wiki/Hamming_distance
```
Comment (Panquesito7) [inline misc/hamming_distance.c:3]:
```suggestion
 * @brief [Hamming distance](https://en.wikipedia.org/wiki/Hamming_distance)
 * algorithm implementation.
```

File: misc/hamming_distance.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+{
+    char str1[] = "karolin";
+    char str2[] = "kathrin";
+
+    assert(hammingDistance(str1, str2) == 3);
+
+    char str3[] = "00000";
+    char str4[] = "11111";
+
+    assert(hammingDistance(str3, str4) == 5);
```
Comment (Panquesito7) [inline misc/hamming_distance.c:49]:
```suggestion
    assert(hammingDistance(str3, str4) == 5);
    printf("All tests have successfully passed!\n");
```

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Good work! 👍
Please enable [GitHub Actions](https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions) in your repository of this fork in this link: https://github.com/aybarsnazlica/C/actions

File: misc/hamming_distance.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+{
+    int i = 0, distance = 0;
+
+    while (str1[i] != '\0')
+
+        if (str1[i] == str2[i])
+        {
+            distance++;
+        }
+    i++;
```
Comment (alexpantyukhin) [inline misc/hamming_distance.c:31]: Maybe I'm wrong but probably `i++` should be under the `while` bracket? Or the algo is in infinite loop?

Comment (aybarsnazlica) [reply misc/hamming_distance.c]: Yes, I added brackets to the while loop.

File: misc/hamming_distance.c [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+ * @param param2 string 2
+ * @returns Hamming distance
+ */
+int hammingDistance(char* str1, char* str2)
+{
+    int i = 0, distance = 0;
+
+    while (str1[i] != '\0')
+    {
+        if (str1[i] == str2[i])
```
Comment (CascadingCascade) [inline misc/hamming_distance.c:27]: Is it just me or shouldn't this be str1[i] != str2[i]? Both asserts failed when I tried to run this code
```suggestion
        if (str1[i] != str2[i])
```

Comment (aybarsnazlica) [reply misc/hamming_distance.c]: Right, thanks.

**Comment by alexpantyukhin** [comment]
Looks good. Approved from my side,

File: misc/hamming_distance.c [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ * algorithm implementation.
+ * @details
+ * In information theory, the Hamming distance between two strings of
+ * equal length is the number of positions at which the corresponding symbols
+ * are different.
+ * @author [Aybars Nazlica](https://github.com/aybarsnazlica)
+ */
+
+#include <assert.h>
+#include <stdio.h>
```
Comment (Panquesito7) [inline misc/hamming_distance.c:13]: Please add a one-line description of what the library/header is for (**see the example below**)
```c
#include <stdio.h>     /// for IO operations
#include <assert.h>    /// for assert
```

Comment (aybarsnazlica) [reply misc/hamming_distance.c]: Fixed, thanks.

**Review by Panquesito7 — DISMISSED** [review]
Awesome work. Thanks! 🚀

File: misc/hamming_distance.c [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+#include <assert.h>  /// for assert
+#include <stdio.h>   /// for IO operations
+
+/**
+ * @brief Function to calculate the Hamming distance between two strings
+ * @param param1 string 1
+ * @param param2 string 2
+ * @returns Hamming distance
+ */
+int hammingDistance(char* str1, char* str2)
```
Comment (tjgurwara99) [inline misc/hamming_distance.c:21]:
```suggestion
int hamming_distance(char* str1, char* str2)
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: misc/hamming_distance.c [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+{
+    char str1[] = "karolin";
+    char str2[] = "kathrin";
+
+    assert(hammingDistance(str1, str2) == 3);
+
+    char str3[] = "00000";
+    char str4[] = "11111";
+
+    assert(hammingDistance(str3, str4) == 5);
```
Comment (tjgurwara99) [inline misc/hamming_distance.c:51]: You also need to change the references!

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: misc/hamming_distance.c (hunk 7 above)
Comment (aybarsnazlica) [reply misc/hamming_distance.c]: I fixed it, thanks.

**Review by tjgurwara99 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**